### ISSUE TITLE
Is it a good idea to enforce max file length?

### ISSUE THREAD

**Issue by hendraanggrian** [pinterest/ktlint#2361]
## Expected Rule behavior

Kotlin recommends fewer than a hundred lines of code for files with multiple declarations. However, they are highly ambiguous on the exact number and if it applies to single declaration files.

For example, the code snippets below would be illegal if the entire file length exceeds 1000. 1000 is a safe number if we define "few" as no more than 10.

```kotlin
class Foo {
    // really long implementation
}

fun Foo.bar() {
    // implementation
}
```

However, it would still be fairly easy for a large project to break this rule. I would love to hear your perspectives.

## Additional information

[Checkstyle](https://checkstyle.sourceforge.io/checks/sizes/filelength.html) and [CodeNarc](https://codenarc.org/codenarc-rules-size.html#classsize-rule) have similar rules with default values set to 2000 and 1000, respectively.

- Current version of ktlint: 1.0.1
- Styleguide section: [Source file organization](https://kotlinlang.org/docs/coding-conventions.html#source-file-organization)

**Comment by paul-dingemans** [comment]
I don't think this should be functionality to be provided by a linter/formatter. It really feels more the domain of static code analyzer which also take cyclomatic complexity into account.

If it would provide a lot of value to your project(s), writing a custom rule would be an option as the rule logic would be very simple.
